documentation/restler/MakeTestProxyRecording.md

```diff
@@ -0,0 +1,187 @@
+The recording file will be created in the `.assets` folder in the root of the azure-rest-api-specs repo,
+but must have a unique name to avoid conflicts with other recordings.
+Here we make the name unique by prepending the same path as the `assets.json` file.
+The response of this request will contain an "x-recording-id" header. Save the value in `recording_id` variable.
+
+```sh
+scenarios=specifications/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/1.0/scenarios
+body='{ 
+  "x-recording-file": "'${scenarios}'/recording.json",
+  "x-recording-assets-file": "'${scenarios}'/assets.json"
+}'
+recording_id=$(curl -X POST -s -D - -d ${body} http://localhost:5000/Record/Start | grep 'x-recording-id' | awk '{print $2}' | sed 's/\r$//')
+```
+
+Now you can update the custom dictionary with the recording id and the test-proxy headers.
+
+```sh
+headers='{                                                 
+  "x-recording-upstream-base-uri": ["https://management.azure.com"],
+  "x-recording-id": ["'${recording_id}'"],
+  "x-recording-mode": ["record"]
+}'
+jq ".restler_custom_payload_header = $headers" dict.json > tmp.json && mv tmp.json dict.json 
+```
+
+### Recompile the configuration
+
+Now recompile the configuration to incorporate these changes.
+
+```sh
+dotnet $restler_bin/restler/Restler.dll compile config.json
+```
+
+## Run the tests
+
+Run the tests as describe in [QuickStart](./QuickStart.md#run-the-tests).
+
+## Stop the recording
+
+Once the tests have completed, stop the recording session by sending a `/Record/Stop` request to the test-proxy.
+```
+curl -X POST -D - -H  "x-recording-id: ${recording_id}" -H 'Content-Length: 0' http://localhost:5000/Record/Stop
+```
+
+Return to the terminal window where the test-proxy is running and press `Ctrl+C` to stop the test-proxy.
+
+The recording file can be found in the `.assets` folder.
+
+```sh
+find .assets -name recording.json
+```
+
+## Upload the Recording
+
+Check the recording file to ensure that all secrets have been removed.
+
+After checking the recordings for secrets, push the recording to the azure-sdk-assets git repository.
+From the root of the azure-rest-api-specs repo, run the command:
+
+```bash
+test-proxy push -a <path to assets.json>
+```
+
+The command will produce output something like this:
+
+```text
+git --version
+git status --porcelain
+git config --get user.name
+git config --get user.email
+git branch 5b23bf70
+git checkout 5b23bf70
+git add -A .
+git -c user.name="Mike Kistler" -c user.email="mikekistler@microsoft.com" commit -m "Automatic asset update from test-proxy."
+git rev-parse --short=10 HEAD
+git remote -v
+git remote set-url origin https://github.com/Azure/azure-sdk-assets
+git tag apitest/appconfiguration/data-plane_2f1f7fc0aa
+git push origin apitest/appconfiguration/data-plane_2f1f7fc0aa
+git remote -v
+git remote set-url origin https://github.com/Azure/azure-sdk-assets
+```
```

Comment:
no need to add the output of test-proxy in this doc? as it contains user name and email.